src/clusterfuzz/_internal/build_management/build_manager.py

```diff
@@ -436,10 +434,74 @@ def _unpack_build(self, base_build_dir, build_dir, build_url):
       storage.copy_file_from(build_url, build_local_archive)
     except Exception as e:
       logs.error(f'Unable to download build from {build_url}: {e}')
-      return False
+      raise
 
     try:
       with build_archive.open(build_local_archive) as build:
+        yield build
+    finally:
+      shell.remove_file(build_local_archive)
+
+  def _open_build_archive(self, unpack_everything: Optional[bool],
```

Comment:
Let's document `unpack_everything` along with other args.

It also feels a bit weird to pass `unpack_everything` as the first argument - I would have put it last but that's just taste.